### PR TITLE
aws_ssm - Rework (and enable) integration tests

### PR DESCRIPTION
**Pull request by tremble**
Depends-On: https://github.com/ansible-collections/community.aws/pull/558

##### SUMMARY

aws_ssm tests have been broken for a while.  This *should* get initial integration tests up and running

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

aws_ssm

##### ADDITIONAL INFORMATION

Notes:
- Does not cover cross-geo S3 buckets (only one region in CI at the minute)
- Does not cover encrypted buckets (missing permissions)

### CODE REVIEW

**Comment by tremble** [comment]
recheck

File: tests/integration/targets/setup_connection_aws_ssm/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            #!/bin/sh
+            sudo dnf install -y https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm
+            sudo systemctl start amazon-ssm-agent
+        os_type: linux
+    amazon:
+        owner: amazon
+        name: amzn2-ami-kernel-5.10-hvm-*-x86_64-gp2
+        user_data: |
+            #!/bin/sh
+            # Pre-Installed ?
```
Comment (markuman) [inline tests/integration/targets/setup_connection_aws_ssm/defaults/main.yml:18]: >             # Pre-Installed ?
afaik yes

Comment (tremble) [reply tests/integration/targets/setup_connection_aws_ssm/defaults/main.yml:18]: Lol, yes it is, that was from when I was still getting the first rework written.  I'll remove the ?

Comment (markuman) [reply tests/integration/targets/setup_connection_aws_ssm/defaults/main.yml:18]: yes. let's reduce confusion

Comment (tremble) [reply tests/integration/targets/setup_connection_aws_ssm/defaults/main.yml:18]: fixed in my local copy, will push once test run finishes

File: tests/integration/targets/setup_connection_aws_ssm/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    - name: Set facts with latest AMIs
+      vars:
+        latest_ami: '{{ ec2_amis.images | sort(attribute="creation_date") | last }}'
+      set_fact:
+        latest_ami_id: '{{ latest_ami.image_id }}'
+
+    # (Local installation of the SSM **client** which is then used by the plugin)
+    - name: Install Session Manager Client for Debian/Ubuntu
+      include_tasks: debian.yml
+      when: ansible_distribution == "Ubuntu" or ansible_distribution == "Debian"
```
Comment (markuman) [inline tests/integration/targets/setup_connection_aws_ssm/tasks/main.yml:44]:
```suggestion
      when: ansible_distribution in ["Ubuntu",  "Debian"]
```

same construct like for rpm based as below. but that's not a blocker

File: tests/integration/targets/setup_connection_aws_ssm/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+    - name: Ensure IAM instance role exists
+      iam_role:
+        name: "ansible-test-{{tiny_prefix}}-aws-ssm-role"
+        assume_role_policy_document: "{{ lookup('file','ec2-trust-policy.json') }}"
+        state: present
+        create_instance_profile: yes
+        managed_policy:
+        - AmazonSSMManagedInstanceCore
+      register: role_output
```
Comment (markuman) [inline tests/integration/targets/setup_connection_aws_ssm/tasks/main.yml:22]:
```suggestion
    - name: Ensure IAM instance role exists
      iam_role:
        name: "ansible-test-{{tiny_prefix}}-aws-ssm-role"
        assume_role_policy_document: "{{ lookup('file','ec2-trust-policy.json') }}"
        state: present
        create_instance_profile: yes
        managed_policy:
        - AmazonSSMManagedInstanceCore
        wait: true
      register: role_output
```

File: tests/integration/targets/setup_connection_aws_ssm/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+          when:
+            - install_plugin_debian is skipped
+            - install_plugin_redhat is skipped
+      always:
+        - debug:
+            var: ansible_distribution
+
+    - name: Wait for IAM Role creation
+      pause:
+        seconds: 10
```
Comment (markuman) [inline tests/integration/targets/setup_connection_aws_ssm/tasks/main.yml:65]:
```suggestion
```

if `wait: true` is set for `iam_role`, this task can be removed.

**Review by markuman — APPROVED** [review]

File: tests/integration/targets/setup_connection_aws_ssm/tasks/main.yml (hunk 3 above)
Comment (markuman) [reply tests/integration/targets/setup_connection_aws_ssm/tasks/main.yml:22]: `wait: true`

**Pull request merged**